### PR TITLE
Adding generic CI/CD integration

### PR DESCRIPTION
**Pull request by haresh-suresh**
Adding generic CI/CD integration
- Removed -g & -d option since we compute image digest by default

### CODE REVIEW

File: inline_scan.sh [hunk 1]
```diff
@@ -181,7 +177,7 @@ get_and_validate_analyzer_options() {
         display_usage_analyzer >&2
         exit 1
     elif ! curl -k -s --fail -H "Authorization: Bearer ${SYSDIG_API_TOKEN}" "${SYSDIG_SCANNING_URL}/policies" > /dev/null; then
-        printf '\n\t%s\n\n' "ERROR - invalid sysdig secure endpoint provided - ${SYSDIG_SCANNING_URL}" >&2
+        printf '\n\t%s\n\n' "ERROR - invalid combintaion of sysdig secure endpoint : token provided - ${SYSDIG_SCANNING_URL} : ${SYSDIG_API_TOKEN}" >&2
```
Comment (marojor) [inline inline_scan.sh]: Minor typo in `combintaion`.

File: inline_scan.sh [hunk 2]
```diff
@@ -195,12 +191,8 @@ get_and_validate_analyzer_options() {
...
             exit 1
         fi
-    elif [[ ! "${g_flag:-}" ]] && [[ ! "${d_flag:-}" ]] && [[ ! "${m_flag:-}" ]]; then
-        printf '\n\t%s\n\n' "ERROR - must provide an image digest, manifest, or specify -g to generate a digest" >&2
-        display_usage_analyzer >&2
-        exit 1
-    elif [[ "${g_flag:-}" ]] && ([[ "${d_flag:-}" ]] || [[ "${m_flag:-}" ]]); then
-        printf '\n\t%s\n\n' "ERROR - cannot generate digest when a manifest or digest is provided" >&2
+    elif [[ "${m_flag:-}" ]]; then
+        printf '\n\t%s\n\n' "ERROR - cannot generate digest when a manifest is provided" >&2
```
Comment (marojor) [inline inline_scan.sh]: If I'm not missing something, providing `-m` option now triggers always an error, so this must be fixed.

File: inline_scan.sh [hunk 3]
```diff
@@ -408,9 +400,15 @@ start_vuln_scan() {
...
     # Prepare commands for container creation & copying all files to container.
-    if [[ "${d_flag-""}" ]]; then
-        CREATE_CMD+=('-d "${IMAGE_DIGEST_SHA}"')
-    fi
+    
+    for i in "${SCAN_IMAGES[@]}"; do
+        # Fetch the individual digest for each succesful image and add it to the list of digests
+        IMAGE_DIGEST=$(docker image inspect "$i" -f "{{.RepoDigests}}" | cut -f2 -d "@" | cut -f1 -d "]")
+        IMAGE_DIGEST_SHA=$IMAGE_DIGEST        
+    done
```
Comment (marojor) [inline inline_scan.sh:274]: If only one image can be analyze at a given time, wouldn't make sense to replace the loop to use `${SCAN_IMAGES[0]}`

File: inline_scan.sh [hunk 4]
```diff
@@ -469,14 +472,31 @@ start_analysis() {
...
+    # Then fetching the result of each scanned digest
+    for ((i=0;  i<${RETRIES}; i++)); do
+        status=$(curl -s -k  --header "Content-Type: application/json" -H "Authorization: Bearer ${SYSDIG_API_TOKEN}" "${SYSDIG_SCANNING_URL}/images/${IMAGE_DIGEST}/checkSummary?tag=$FULLTAG" | grep "status" | awk '{print $2}')
+        if [ ! -z  "$status" ]; then
+            break
         fi
+        echo -n "." && sleep 5
+    done
+ 
+    echo -n "Scan Report - "
```
Comment (marojor) [inline inline_scan.sh]: **nit**: I would print the new line here, so it's more convenient to copy the JSON:

```
Scan Report - {
  "imageDigest": "sha256:72c42ed48c3a2db31b7dafe17d275b634664a708d901ec9fd57b1529280f01fb",
  "at": "2019-10-10T08:24:15Z",
  "tag": "alpine",
  "status": "pass",
```

File: inline_scan.sh [hunk 5]
```diff
@@ -469,14 +472,31 @@ start_analysis() {
...
+
+    FULLTAG="${SCAN_IMAGES[0]}"
+    check_status_with_digest
+}
+
+check_status_with_digest() {
+    RETRIES=100
+    # Then fetching the result of each scanned digest
+    for ((i=0;  i<${RETRIES}; i++)); do
+        status=$(curl -s -k  --header "Content-Type: application/json" -H "Authorization: Bearer ${SYSDIG_API_TOKEN}" "${SYSDIG_SCANNING_URL}/images/${IMAGE_DIGEST}/checkSummary?tag=$FULLTAG" | grep "status" | awk '{print $2}')
```
Comment (marojor) [inline inline_scan.sh]: This command will keep the last comma and quotes in `$status`, making the check later to return always fail:

```
➜  secure-inline-scan ✗ curl -s -k --header "Content-Type: application/json" -H "Authorization: Bearer c116da15-4d54-4f05-90e1-19633c389394" https://secure-staging.sysdig.com/api/scanning/v1/images/sha256:72c42ed48c3a2db31b7dafe17d275b634664a708d901ec9fd57b1529280f01fb/checkSummary\?tag\=alpine | grep "status" | awk '{ print $2 }'
"pass",
```

The simplest I could come up with to fix this is:

```
...| grep "status" | cut -d : -f 2 | awk -F\" '{ print $2 }'
```
 maybe there's something simpler.

**Review by marojor — CHANGES_REQUESTED** [review]
A bug must be fixed while checking for the scan report status. 

Some other minor fixes suggested.

File: inline_scan.sh (hunk 3 above)
Comment (haresh-suresh) [reply inline_scan.sh:274]: This is true, but this is part of a larger refactor that we will need to do. So leaving this alone in this PR, but will address this in a separate ticket/PR at a later date.

**Review by yathi-sysdig — APPROVED** [review]

**Pull request merged**